code/components/extra-natives-rdr3/include/GamePrimitives.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+// SSE4 is pretty common
+#define _XM_SSE4_INTRINSICS_
+#include <DirectXMath.h>
+
+namespace rage
+{
+struct alignas(16) Vec3V
+{
+	float x;
+	float y;
+	float z;
+	float pad;
+
+	Vec3V()
+		: x(0), y(0), z(0), pad(0)
+	{
+	}
+
+	Vec3V(float x, float y, float z)
+		: x(x), y(y), z(z), pad(NAN)
```

Comment:
I don't think relying on this padding for anything related to 4x4 matrices is a good idea. I believe the game might fill this field with some trash during some SSE optimized calculations. Plus this code was copied from `extra-natives-five`, i.e. already was like that for many years. I think we should make some kind of a shared component/library for all RAGE games with the stuff that unlikely to be changed anytime soon. But this is definitely not a subject for this PR.

Reply:
Doing things for no good reason in the past, is not a reason to not do it correct in the future. Setting it to `1.0` by default will result in correct behavior may the game actually use the `.w` with matrix calculation (SSE or not), `0.0` and `NaN` do not.

Reply:
I don't think using 1.0f can be considered as "correct behavior", these 4 bytes are padding, you shouldn't consider this as "w", it's very often filled with some trash in game's memory. It shouldn't be used in an any unexpected way, it would pretty obviously lead to an undefined behavior as we can't exactly know what the would could do with this padding. If you need "w" component you should use Vec4V instead.

Reply:
It is in the case of, like I mentioned: "may the game (or us) actually use the .w"

Reply:
The game won't ever "use it" in an any reasonable way, when R* want to store position (for instance) and use "w" component for something else, they're using Vec4V. Here's an example: `rage::Vec4V m_BoundingBoxMaxXYZMarginW;`  from `rage::phBound`. I was trying to say that "w" is *not* used and but might filled with some unusable trash because of the calculation optimization algorithms, we shouldn't rely on this, we shouldn't use `Vec3V` if we need an access to "w" component, that's just stupid, it's just a padding to make it easier for certain calculations and ensure better memory layout (i.e. 16 bytes struct).

Reply:
I'm not saying we're relying on this, nor should we. I know why the padding is there. The issue is when we use this vec3 at a place where some system expects it to be a vec4 (and we don't know), putting it to 1.0 would allow those systems to calculate normally. For us that's a simple change

Reply:
I don't think there was such cases before. It's usually kinda easy to detect if the game wants Vec3 or Vec4 in decompiled code. Moreover, it would be better to get attention of such cases by getting weird calculations rather than not being aware and hopping everything would work as expected.

Reply:
Good point, not sure what value would be obvious, maybe `NaN` them both then

Reply:
> putting it to 1.0 would allow those systems to calculate normally.

Unless it's a normal (: 

To supplement, `NaN` would be nice because it propagates (making errors especially obvious).

Reply:
> The issue is when we use this vec3 at a place where some system expects it to be a vec4 (and we don't know), putting it to 1.0 would allow those systems to calculate normally. 

If, in these cases, this would be something that's _not_ meant to be `1.0`, you can't know. I second @Disquse saying that `NaN` at least would be _noticeably_ leading to wrong outputs due to propagation.